services/cloudrun-service/pkg/cloudrun/cloudrun_test.go

```diff
@@ -0,0 +1,307 @@
+)
+
+const errorMessage = "Expected %s, but got %s"
+
+func TestIsPresent(t *testing.T) {
+	for _, test := range []struct {
+		testName         string
+		servicesResponse []*run.Service
+		serviceName      string
+		expectedResponse bool
+	}{
+		{
+			testName:         "No services returned",
+			servicesResponse: []*run.Service{},
+			serviceName:      "myService",
+			expectedResponse: false,
+		},
+		{
+			testName: "Service does not exist",
+			servicesResponse: []*run.Service{
+				{Metadata: &run.ObjectMeta{Name: "notMyService"}},
+				{Metadata: &run.ObjectMeta{Name: "alsoNotMyService"}},
+				{Metadata: &run.ObjectMeta{Name: "notIt"}},
+			},
+			serviceName:      "myService",
+			expectedResponse: false,
+		},
+		{
+			testName: "Service exists",
+			servicesResponse: []*run.Service{
+				{Metadata: &run.ObjectMeta{Name: "notMyService"}},
+				{Metadata: &run.ObjectMeta{Name: "myService"}},
+				{Metadata: &run.ObjectMeta{Name: "notIt"}},
+			},
+			serviceName:      "myService",
+			expectedResponse: true,
+		},
+	} {
+		testCase := test
+		t.Run(testCase.testName, func(t *testing.T) {
+			t.Parallel()
+			exists := isPresent(testCase.servicesResponse, testCase.serviceName)
+			if exists != testCase.expectedResponse {
+				t.Errorf("Expected %t, but got %t", testCase.expectedResponse, exists)
+			}
+		})
+	}
+}
+
+func TestGetParent(t *testing.T) {
+	for _, test := range []struct {
+		testName       string
+		service        *run.Service
+		expectedParent string
+		expectedError  error
+	}{
+		{
+			testName: "Namespace does not exist",
+			service: &run.Service{
+				Metadata: &run.ObjectMeta{
+					Name:   "testService",
+					Labels: map[string]string{serviceLocationLabel: "europe-west1"},
+				},
+			},
+			expectedParent: "",
+			expectedError:  serviceConfigError{name: "testService", namespaceMissing: true},
+		},
+		{
+			testName: "Location does not exist",
+			service: &run.Service{
+				Metadata: &run.ObjectMeta{Name: "testService", Namespace: "proj_id"},
+			},
+			expectedParent: "",
+			expectedError:  serviceConfigError{name: "testService", locationMissing: true},
+		},
+		{
+			testName: "Namespace and Location do not exist",
+			service: &run.Service{
+				Metadata: &run.ObjectMeta{Name: "testService"},
+			},
+			expectedParent: "",
+			expectedError:  serviceConfigError{name: "testService", namespaceMissing: true},
+		},
+		{
+			testName: "Namespace and Location do exist",
+			service: &run.Service{
+				Metadata: &run.ObjectMeta{
+					Name:      "testService",
+					Namespace: "proj_id",
+					Labels:    map[string]string{serviceLocationLabel: "europe-west1"},
+				},
+			},
+			expectedParent: "projects/proj_id/locations/europe-west1",
+			expectedError:  nil,
+		},
+	} {
+		testCase := test
+		t.Run(testCase.testName, func(t *testing.T) {
+			t.Parallel()
+			parent, err := getParent(testCase.service)
+			if err != nil {
+				if diff := cmp.Diff(testCase.expectedError, err, cmpopts.EquateErrors()); diff != "" {
```

Comment:
```suggestion
				if diff := cmp.Diff(testCase.expectedError, err, cmpopts.EquateErrors()); diff != "" {
```
I think the first diff is redundant, because cmp.Diff will handle it.